scrapper.py

```diff
@@ -25,39 +39,151 @@ class Crawler:
+                       'июля': '07',
+                       'августа': '08',
+                       'сентября': '09',
+                       'октября': '10',
+                       'ноября': '11',
+                       'декабря': '12',
+                       }
+        for month in months_dict:
+            if month in raw_date:
+                raw_date = raw_date.replace(month, months_dict[month])
+            elif month.capitalize() in raw_date:
+                raw_date = raw_date.replace(month.capitalize(), months_dict[month])
+        self.article.date = datetime.strptime(raw_date, '%d %m %Y %H:%M')
+        # TOPICS
+        topics_bs = article_bs.find('div', class_='entry__tags').find_all('a')
+        for topic_bs in topics_bs:
+            self.article.topics.append(topic_bs.text[1:])
+
+    def _fill_article_with_text(self, article_bs):
+        text_of_article = ''
+        texts = article_bs.find('article', class_='entry__body').find_all('p')
+        for text_part in texts:
+            text_of_article += text_part.text
+        self.article.text = text_of_article
+
+    def parse(self):
+        response = requests.get(url=self.article_url)
+        article_bs = BeautifulSoup(response.text, 'lxml')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    path = Path(base_path)
+    if path.exists():
+        shutil.rmtree(path)
+    path.mkdir(parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as file:
+        config = json.load(file)
+
+    if not isinstance(config["seed_urls"], list):
+        raise IncorrectURLError
+
+    for url in config["seed_urls"]:
+        if HTTP_PATTERN not in url:
+            raise IncorrectURLError
+
+    seed_urls = config["seed_urls"]
+    max_articles = config["total_articles_to_find_and_parse"]
+
+    if not seed_urls:
+        raise IncorrectURLError
+
+    if not isinstance(max_articles, int) or max_articles <= 0:
+        raise IncorrectNumberOfArticlesError
+
+    if max_articles > 200:
+        raise NumberOfArticlesOutOfRangeError
+
+    return seed_urls, max_articles
 
 
 if __name__ == '__main__':
-    # YOUR CODE HERE
-    pass
+    test_seed_urls, test_max_articles = validate_config(CRAWLER_CONFIG_PATH)
+    prepare_environment(ASSETS_PATH)
+
+    print('craw')
+    crawler = Crawler(test_seed_urls, test_max_articles)
+    crawler.find_articles()
+
+    print('pars')
+    for id_of_article, article_url_test in enumerate(crawler.urls):
+        article_parser = HTMLParser(article_url=article_url_test, article_id=id_of_article)
+        article = article_parser.parse()
+        article.save_raw()
+        print(id_of_article,'done')
+        id_of_article += 1
```

Comment:
you don't have to add one after each iteration: it is going to happen automatically as _i_ in `for i, j in enumerate(iterable)` will progressively take value from 0 to `len(iterable) - 1`
This is why you should also make sure you do not accidentally create files with 0 as their id

Reply:
have you? 
there is still `id_of_article += 1` right before its reassignment

Reply:
now i did